hack/generated/pkg/genruntime/property_bag.go

```diff
@@ -7,25 +7,39 @@ package genruntime
 
 import (
 	"encoding/json"
+
 	"github.com/pkg/errors"
 )
 
-// PropertyBag is used to stash additional information that's not directly supported by storage
+// PropertyBag is an unordered set of stashed information that used for properties not directly supported by storage
 // resources, allowing for full fidelity round trip conversions
 type PropertyBag map[string]string
 
-// Clone creates an independent copy of the property bag
-func (bag PropertyBag) Clone() PropertyBag {
+// PropertyBag returns a new property bag
+// originals is a (potentially empty) sequence of existing property bags who's content will be copied into the new
+// property bag. In the case of key overlaps, the bag later in the parameter list overwrites the earlier value.
+func NewPropertyBag(originals ...PropertyBag) PropertyBag {
 	result := make(PropertyBag)
-	for k, v := range bag {
-		result[k] = v
+
+	for _, orig := range originals {
+		for k, v := range orig {
+			result[k] = v
+		}
 	}
 
 	return result
 }
 
+// Contains returns true if the specified name is present in the bag; false otherwise
+func (bag PropertyBag) Contains(name string) bool {
+	_, found := bag[name]
+	return found
+}
+
 // Add is used to add a value into the bag; exact formatting depends on the type.
 // Any existing value will be overwritten.
+// property is the name of the item to put into the bag
+// value is the instance to be stashed away for later
 func (bag PropertyBag) Add(property string, value interface{}) error {
 	switch v := value.(type) {
 	case string:
```

Comment:
minor: This isn't in the code you actually changed, but I'm curious why `string` is called out here but not other primitive types?

Feels like you could probably get away without any special casing and just always use `json.Marshal`? I am assuming that `json.Marshal(str)` just returns `str` but I don't know for sure.

Reply:
I don't recall - might have been to avoid the overhead of calling `json.Marshal` when not required, or it may have been to handle round trip formatting issues. I do remember it was deliberate though.

Reply:
Might be worth leaving a comment there as to the reason? Especially if there was some problem if it wasn't handled specially?

Reply:
I have a laundry list of minor code-gardening things to do post `codegen-alpha-0`, I'll add this to that.